### ISSUE TITLE
Improve RBAC rules for the case of several ceph clusters in one kubernetes cluster

### ISSUE THREAD

**Issue by dimm0** [rook/rook#7427]
**Is this a bug report or feature request?**
* Feature Request

**What should the feature do:**

Currently upgrade and install instructions are assuming a single ceph cluster. Rook supports the case of several ceph clusters in different namespaces inside a single kubernetes cluster. The upgrade and install instructions and RBAC yamls should support this case.

**Comment by travisn** [comment]
We need to look into Kustomize or similar tools for generating the manifests for multiple clusters.

**Comment by BlaineEXE** [comment]
Yes. This will be a good item to think about for upcoming Helm work also. https://github.com/rook/rook/issues/6776

**Comment by dimm0** [comment]
🥺

**Comment by travisn** [comment]
The new [Cluster helm chart](https://rook.github.io/docs/rook/v1.6/helm-ceph-cluster.html) supports installing clusters in different namespaces with the needed RBAC. 

If there is a cephcluster CR that is not managed by helm, it seems helm can take over the CR. This would allow future management of the RBAC to be handled by the helm chart. See #8136. 

@dimm0 Would that cover it? Or still need instructions for the non-helm case?

**Comment by dimm0** [comment]
I don't need instructions, I just sit with a text editor, and I'm done in about an hour of heavy editing :)

And no, I'm not comfortable using helm on this..

Ok, I guess won't fix then.

**Comment by travisn** [comment]
Actually what about [these instructions](https://rook.github.io/docs/rook/v1.6/ceph-advanced-configuration.html#deploying-a-second-cluster)? Not too long ago we added the `common-second-cluster.yaml` to help with CI scenarios for multiple clusters, so it should be reliable.

**Comment by dimm0** [comment]
Ok, have to check. How is it dealing with addressing multiple namespaces from the main operator one? That was the issue before. Will it create multiple roles for each such namespace?

**Comment by travisn** [comment]
Right, it does create different bindings for each namespace instead of appending to existing bindings, so there is still that issue. Looks like it also has a cluster role binding that would be an issue for multiple namespaces.

**Comment by dimm0** [comment]
Soo.. Leave the ticket?

I see a simple solution by moving such role bindings to a separate file, so that I don't have to search for them at least

**Comment by travisn** [comment]
Seems like we need a script that would handle this case. Since the most common case is for a single cluster, I wouldn't want to move them to a new file.

**Comment by dimm0** [comment]
> Actually what about [these instructions](https://rook.github.io/docs/rook/v1.6/ceph-advanced-configuration.html#deploying-a-second-cluster)? Not too long ago we added the `common-second-cluster.yaml` to help with CI scenarios for multiple clusters, so it should be reliable.

I don't see these https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/common.yaml#L594-L622 rolebindings addressed in 2nd cluster file

**Comment by travisn** [comment]
Looks like the osd role binding is named `rook-ceph-osd-external` instead of `rook-ceph-osd`. That really shouldn't be necessary though, but explains why it should basically be working.
The `rook-ceph-mgr` binding IIRC should only be need when using the `rook` mgr module of ceph, which we have disabled by default. Since it's not critical, things should work fine without it, although we should certainly fix it to be consistent.
